build-scripts/packageExtension.js

```diff
@@ -0,0 +1,118 @@
+/*!
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+const fs = require('fs-extra')
+const path = require('path')
+const ncp = require('child_process')
+const shell = require('shelljs')
+
+const timeMessage = 'Packaged extension'
+const manifestFile = 'vss-extension.json'
+const tasksDirectory = 'Tasks'
+const repoRoot = path.dirname(__dirname)
+const inTasks = path.join(repoRoot, tasksDirectory)
+const outBuildTasks = path.join(repoRoot, '_build', tasksDirectory)
+const outPackage = path.join(repoRoot, '_package')
+const outPackageTasks = path.join(outPackage, tasksDirectory)
+
+const unprocessFolders = [
+    'Common',
+    '.DS_Store'
+]
+
+function findMatchingFiles(directory) {
+    return fs.readdirSync(directory)
+}
+
+
+function package(options) {
+    fs.mkdirpSync(outPackage)
+
+    fs.copySync(path.join(repoRoot, 'LICENSE'), path.join(outPackage, 'LICENSE'), {overwrite: true})
+    fs.copySync(path.join(repoRoot, 'README.md'), path.join(outPackage, 'README.md'), {overwrite: true})
+    fs.copySync(path.join(repoRoot, '_build', manifestFile), path.join(outPackage, manifestFile), {overwrite: true})
+
+    // stage manifest images
+    fs.copySync(path.join(repoRoot, 'images'), path.join(outPackage, 'images'), {overwrite: true})
+
+    fs.mkdirpSync(outPackageTasks)
+
+    // clean, dedupe and pack each task as needed
+    findMatchingFiles(inTasks).forEach(function(taskName) {
+        console.log('Processing task ' + taskName)
+
+        if(!unprocessFolders.every((folderName) => { return folderName !== taskName})) {
+            console.log('Skpping task ' + taskName)
+            return
+        }
+
+        var taskBuildFolder = path.join(outBuildTasks, taskName)
+        var taskPackageFolder = path.join(outPackageTasks, taskName)
+        fs.mkdirpSync(taskPackageFolder)
+
+        var taskDef = require(path.join(taskBuildFolder, 'task.json'))
+        if (!taskDef.execution.hasOwnProperty('Node')) {
+            console.log('Copying non-node task ' + taskName)
+
+            fs.copySync(taskBuildFolder, taskPackageFolder)
+            return
+        }
+        shell.cd(taskBuildFolder)
+        fs.copySync(path.join(taskBuildFolder, 'task.json'), path.join(taskPackageFolder, 'task.json'), {overwrite: true})
```

Comment:
Looks like you could create an array of src/dst objects here, then iterate on that array with fs.copySync to reduce verbosity.

Reply:
sure, added an array